doc/project/release_process.rst

```diff
@@ -85,24 +85,155 @@ At that point, the whole process starts over again.
 
 .. _release_quality_criteria:
 
-Release Quality Criteria
-************************
+Release Criteria
+****************
+
+The main motivation is to clearly have the criteria in place that must be met for a public releases to ship.
+This will help define when a release is "done" in terms that most people can
+understand and in ways that help new people to understand the process and
+participate in creating successful releases:
+
+- The release criteria documents all the requirements of our target audience for each Zephyr release
+- The target audiences for each public release can be different, and may overlap
+- The criteria at any given time are not set in stone: there may be requirements
+  that have been overlooked, or that are new, and in these cases, the criteria
+  should be expanded to ensure all needs are covered.
+
+Below is the high level criteria to be met for each release:
+
+- No blocker bugs
+- All relevant tests shall pass on Tier 0 platforms
+- A test subset shall pass on Tiers 0 and 1 (at least 1 per architecture/architecture variant/Hardware features)
+- All samples/tests shall build on Tiers 0, 1 and 2
+- All high and critical static analysis and security issues addressed
+- Footprint/Performance Tracking (no specific KPIs)
+- License audits pass
+- Documentation is complete and Release Notes are up-to-date.
+- Per Subsystem release criteria + Testing
+- Reproducible Builds
+
+Bug Severity and Priority
+=========================
+
+Priority of bugs is not enough to make release decisions: not all critical bugs are of highest
+priority and they do not need to be addressed in a release. Priority can also be
+set high on less critical bugs.
+
+Severity shall be used to classify bugs based on impact to users and community
+in general. For example:
+
+- A kernel bug impacting every user can be set as critical or major. If it needs
+  to be fixed in the next release, it will need to be marked as a blocker bug.
+- A Kconfig option or DTS in a board is impacting a small group of users, so it
+  could be minor or moderate but, a bug in hardware that is marked as a
+  `reference hardware` can be major or critical and in some cases a release
+  blocker.
+
+Bug priorities can be used by owners of bugs in an area or subsystem to manage
+their workflow and priorities.
+A low or moderate bug in a platform might be set to `High Priority` by the
+owner to manage execution within a subsystem or by a hardware vendor.
```

Comment:
Should we just remove priority from our documentation and just document severity/impact with the additional modifier of "blocker" for issues that we deem needing a fix before release.

Reply:
Based on the discussions we had, priority can be used by teams and maintainers.  We should leave this at least for now and see how things will work out and can be removed later.